docs/reference/mapping.asciidoc

```diff
@@ -5,30 +5,54 @@
 --
 
 Mapping is the process of defining how a document, and the fields it contains,
-are stored and indexed. For instance, use mappings to define:
+are stored and indexed.
 
-* which string fields should be treated as full text fields.
-* which fields contain numbers, dates, or geolocations.
-* the <<mapping-date-format,format>> of date values.
-* custom rules to control the mapping for
-  <<dynamic-mapping,dynamically added fields>>.
+Each document is a collection of fields, which each have their own
+<<mapping-types,data type>>. When mapping your data, you create a mapping
+definition, which contains a list of fields or `properties` that are pertinent
+to the document. A mapping definition also includes
+<<mapping-fields,metadata fields>>, like the `_source` field, which customize
+how a document's associated metadata is handled.
+
+You can choose a _dynamic mapping_ or an _explicit mapping_ to define your data.
+Each method provides different benefits based on where you are in your data
+journey.
 
-A mapping definition includes metadata fields and fields:
+NOTE: Before 7.0.0, the mapping definition included a type name.
+{es} 7.0.0 and later no longer accept a _default_ mapping. See <<removal-of-types>>.
 
-<<mapping-fields,Metadata fields>>::
+[discrete]
+[[mapping-dynamic]]
+== Dynamic mapping
+<<dynamic-field-mapping,Dynamic mapping>> allows you to experiment with
+and explore data when you’re just getting started. {es} adds new fields
+automatically, just by indexing a document. You can add fields to the top-level
+mapping, and to inner <<object,`object`>>  and <<nested,`nested`>> fields.
 
-Metadata fields are used to customize how a document's associated metadata is
-treated. Examples of metadata fields include the document's
-<<mapping-index-field,`_index`>>, <<mapping-id-field,`_id`>>, and
-<<mapping-source-field,`_source`>> fields.
+You can also dynamically map runtime fields:
 
-<<mapping-types,Fields>>::
+* <<match-mapping-runtime-fields,Map runtime fields in a dynamic template>> so
+that new fields are automatically added to the index mapping as runtime fields.
+* <<runtime-search-request,Define runtime fields in a search request>> to
+experiment with different mapping options, and also fix mistakes in your index
+mapping values by overriding values in the mapping during the search request.
```

Comment:
I would not mention here runtime fields explicitly, as they should be considered just fields. It should not come as a surprise that runtime fields can be both mapped explicitly or dynamically

Reply:
Agreed, though I'm wondering if we want to mention dynamic templates here. 

>Dynamic mapping allows you to experiment with and explore data when you’re just getting started. Elasticsearch adds new fields automatically, just by indexing a document. You can add fields to the top-level mapping, and to inner object and nested fields.

>Use dynamic templates to define custom mappings that are applied to dynamically added fields based on the matching condition. You can also map runtime fields in a dynamic template so that new fields are automatically added to the index mapping as runtime fields.

Reply:
To me, dynamic templates belong to the dynamic mapping side of things. I am not sure if I would mention them explicitly here.